### DISCUSSION TITLE
Predicates on relationship tags

### DISCUSSION THREAD

**Discussion by susliko** [likec4/likec4#536] [General]
[View predicates](https://likec4.dev/docs/dsl/views/#view-predicates) is a really nice feature.

I think it might be useful to allow these on relationship predicates. This would enable [c4 dynamic-diagrams](https://c4model.com/#DynamicDiagram), e.g:

```
specification {
  tag pay
}

model {
  // `my-online-store` system is already defined somewhere
  extend my-online-store {
     front -> back "1. Make a payment of this cart" #pay 
     back -> payments "2. Initialize payment request" #pay
     back -> front "3. Redirect to payment form" #pay 
     payment-form -> payments "4. process card data" #pay
     payments -> back "5. money blocked" #pay 
     back -> payments "6. confirm" #pay
  }
}

views {
  view payment {
    #pay
    include my-online-store, payments
    include relationship.tag = #pay 
 }
}
```

**Comment by mcpride** [comment]
Yes, filters (include, exclude) for relationships e.g. by tags would be very helpful!

**Comment by davydkov** [comment]
There were [#593](https://github.com/likec4/likec4/issues/593)
```zig
views {
  view payment {
    #pay
    include my-online-store, payments
    include relationship.tag = #pay 
  }
}
```

At the moment, predicate `include my-online-store, payments` will include any relationships between these elements.
So, as workaround, first we need to exclude all relationships, and then include `#pay`?

btw, I'm working on dynamic views

**Reply by susliko** [reply]
>  So, as workaround, first we need to exclude all relationships, and then include #pay?

Yes, it could definitely be a workaround if tag predicates could be applied to relationships.

> btw, I'm working on dynamic views

It's great to hear!

**Comment by davydkov** [comment]
Solved with [`where`](https://likec4.dev/dsl/views/#filter) predicate
